pkg/elasticache.go

```diff
@@ -0,0 +1,108 @@
+package pkg
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/app-sre/aws-resource-exporter/pkg/awsclient"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var RedisVersion *prometheus.Desc = prometheus.NewDesc(
+	prometheus.BuildFQName(namespace, "", "elasticache_redisversion"),
+	"The ElastiCache engine type and version.",
+	[]string{"aws_region", "cache_name", "engine", "engine_version", "aws_account_id"},
+	nil,
+)
+
+type ElastiCacheExporter struct {
+	sessions     []*session.Session
+	svcs         []awsclient.Client
+	cache        MetricsCache
+	awsAccountId string
+
+	logger   log.Logger
+	timeout  time.Duration
+	interval time.Duration
+}
+
+// NewElastiCacheExporter creates a new ElastiCacheExporter instance
+func NewElastiCacheExporter(sessions []*session.Session, logger log.Logger, config ElastiCacheConfig, awsAccountId string) *ElastiCacheExporter {
+	level.Info(logger).Log("msg", "Initializing ElastiCache exporter")
+
+	var elasticaches []awsclient.Client
+	for _, session := range sessions {
+		elasticaches = append(elasticaches, awsclient.NewClientFromSession(session))
+	}
+
+	return &ElastiCacheExporter{
+		sessions:     sessions,
+		svcs:         elasticaches,
+		cache:        *NewMetricsCache(*config.CacheTTL),
+		logger:       logger,
+		timeout:      *config.Timeout,
+		interval:     *config.Interval,
+		awsAccountId: awsAccountId,
+	}
+}
+
+func (e *ElastiCacheExporter) getRegion(sessionIndex int) string {
+	return *e.sessions[sessionIndex].Config.Region
+}
+
+// Gets ElastiCache info and adds it to metrics cache
+func (e *ElastiCacheExporter) getElastiCacheInfo(sessionIndex int, clusters []*elasticache.CacheCluster) {
```

Comment:
This is not getting anything, but creating Metrics and adding them to the cache. What do you think about addMetricFromElastiCacheInfo?